examples/c-guest/guest/main.c

```diff
@@ -0,0 +1,34 @@
+#include <stdint.h>
+#include "platform.h"
+
+__asm__(
+	".global _start\n\t"
+	".type _start, @function\n\t"
+	"_start:\n\t"
+	"la sp,      _stack_top\n\t"
+
+	// Load the globals pointer. The program will load pointers relative to this
+	// register, so it must be set to the right value on startup.
+	// See: https://gnu-mcu-eclipse.github.io/arch/riscv/programmer/#the-gp-global-pointer-register
+	// Linker relaxations must be disabled to avoid the initialization beign
+	// relaxed with an uninitialized global pointer: mv gp, gp
+	".option push;\n\t"
+	".option norelax\n\t"
+	"la gp,      __global_pointer$\n\t"
+	".option pop\n\t"
+
+	// Jump to main function
+	"call main\n\t");
```

Comment:
Is there anyway to put this in `platform`?

Reply:
hmm, good thought. Let me try, I assumed not, but not for a good reason